clients/src/main/java/org/apache/kafka/clients/consumer/internals/SharedConsumerState.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.clients.ApiVersions;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.internals.events.CheckAndUpdatePositionsEvent;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.Time;
+
+import java.time.Duration;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class stores shared state needed by both the application thread ({@link AsyncKafkaConsumer}) and the
+ * network thread ({@link ConsumerNetworkThread}) to avoid costly inter-thread communication, where possible.
```

Comment:
nit: We have used "background thread" rather than "network thread" everywhere else. And I know the class name is different, but I would go for consistency in the comments so people do not get confused about a new kind of thread.

Reply:
Out of curiosity I looked up the number of instances of "background thread" vs. "network thread" in the client source:

```
$ grep -R "network thread" clients/src/main/java | wc -l
      61
$ grep -R "background thread" clients/src/main/java | wc -l
      42
```

That's not exact, of course, but there are clearly two warring factions fighting for dominance here 😆